web-wallet/src/lib/components/__tests__/Transactions.spec.js

```diff
@@ -0,0 +1,158 @@
+import { cleanup, render } from "@testing-library/svelte";
+import { afterEach, describe, expect, it, vi } from "vitest";
+import Transactions from "../Transactions/Transactions.svelte";
+import { get } from "svelte/store";
+import { settingsStore } from "$lib/stores";
+
+import { transactions } from "$lib/mock-data";
+import { sortByHeightDesc } from "$lib/transactions";
+
+global.ResizeObserver = vi.fn().mockImplementation(() => ({
+	observe: vi.fn(),
+	unobserve: vi.fn(),
+	disconnect: vi.fn()
+}));
+
+vi.useFakeTimers();
+
+describe("Transactions", () => {
+	const transactionsPromise = Promise.resolve(transactions);
+	const emptyTransactionsPromise = Promise.resolve([]);
+	const blockExplorerBaseUrl = "https://explorer.dusk.network/transactions/transaction?id=";
+	const highestTransactionID = "tg874534tddsghdtst54hdfh795h";
```

Comment:
It's better to avoid hardcoded values here and elsewhere, as they rely on the current mock data which can change in time.
Depending on the test you can pass a promise that resolves to an expected result or retrieve values from the current mock data.